### PR TITLE
[MB-7550] route moves to services counseling

### PR DESCRIPTION
**Pull request by pearl-truss**
## Description

This PR updates the moveRouter services so that submitted moves are queried and routed to the service counselor queue when the origin duty station provides services counseling and otherwise routes to the TOO queue. The updates the previous behavior of having all moves sent to the services counseling queue

## Setup

```sh
make db_dev_e2e_populate && make server_run
make client_run
```

### Verify Moves with Origin Duty Stations w/o Services Counseling are Routed to the TOO Queue
1. Create a move with an origin duty station whose GBLOC matches your local office user and doesn't provide services counseling. I used USMC San Diego
2. Once move is created, sign into the office view as the TOO -- the move should appear in this queue with a status of New Move
3. Switch to the services counseling view -- the move should not appear here

### Verify Moves with Origin Duty Stations with Services Counseling are Routed to the Services Counseling Queue
1. Create a move with an origin duty station whose GBLOC matches your local office user and provides services counseling. I used Presidio of Monterey
2. Once move is created, sign into the office view as the service counselor -- the move should appear in this queue with a status of Needs Services Counseling
3. Switch to the TOO view -- the move should not appear here

## Code Review Verification Steps
* [x ] Request review from a member of a different team.
* [ x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7550) for this change
* [this article](tbd) explains more about the approach used.

### CODE REVIEW

File: cmd/milmove/serve.go [hunk 1]
```diff
@@ -598,7 +598,7 @@ func serveFunction(cmd *cobra.Command, args []string) error {
 	if len(gexURL) == 0 {
 		// this spins up a local test server
 		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(http.StatusOK)
+			w.WriteHeader(http.StatusForbidden)
```
Comment (mboren) [inline cmd/milmove/serve.go:601]: What does this do?

Comment (pearl-truss) [reply cmd/milmove/serve.go:601]: whoops, didnt mean to commit this change

File: pkg/services/move/move_router.go [hunk 2]
```diff
@@ -45,33 +44,41 @@ func (router moveRouter) Submit(move *models.Move) error {
...
+			return false, err
+		}
 	}
-	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
-	v.AutomaticEnv()
 
-	return v.GetBool(cli.FeatureFlagServiceCounseling)
+	var originDutyStation models.DutyStation
+
+	if orders.OriginDutyStationID != nil && *orders.OriginDutyStationID != uuid.Nil {
```
Comment (gidjin) [inline pkg/services/move/move_router.go:66]: Not really part of this PR but I wonder in what situations orders would not have an Origin Duty Station. Which makes me wonder if returning `false` when there isn't one or returning an error is better.

**Review by gidjin — COMMENTED** [review]
Works as expected, code looks good to me. I left a question about the remaining TODO.

File: pkg/services/move/move_router.go [hunk 3]
```diff
@@ -45,33 +44,41 @@ func (router moveRouter) Submit(move *models.Move) error {
 	return nil
 }
 
-// TODO: Replace the code in this function to determine whether or not the move
-// needs service counseling based on the service member's origin duty station.
-// Then remove all code related to the service counseling feature flag here and
-// in pkg/cli/featureflag.go, and remove any references to
-// `FEATURE_FLAG_SERVICE_COUNSELING` from the entire project.
-// You'll need to update the test setup in TestSubmitMoveForServiceCounselingHandler
+// TODO: (Remaining) You'll need to update the test setup in TestSubmitMoveForServiceCounselingHandler
```
[the remark this answers is not in the document]
Comment (pearl-truss) [reply pkg/services/move/move_router.go:47]: I made this change and just forgot to remove this comment

[the remark this answers is not in the document]
Comment (gidjin) [reply pkg/services/move/move_router.go:47]: Sorry I'd missed that update. I'll approve the PR, just remove the TODO before merging please :)

**Review by gidjin — APPROVED** [review]
